src/main/java/net/openhft/chronicle/bytes/internal/BytesInternal.java

```diff
@@ -2522,36 +2523,39 @@ public static double parseDouble(@NotNull StreamingDataInput in)
         boolean negative = false;
         int decimalPlaces = Integer.MIN_VALUE;
         boolean digits = false;
-        int ch;
-        do {
-            ch = in.rawReadByte() & 0xFF;
-        } while (ch == ' ' && in.readRemaining() > 0);
+        int ch = ' ';
+        while (in.readRemaining() > 0) {
+            ch = in.peekUnsignedByte() & 0xFF;
+            if (ch != ' ')
+                break;
+            else
+                in.readSkip(1);
+        }
 
         try {
             switch (ch) {
                 case 'N':
-                    if (compareRest(in, "aN"))
+                    if (compareRest(in, NAN))
                         return Double.NaN;
-                    in.readSkip(-1);
 
                     return Double.NaN;
                 case 'I':
-                    //noinspection SpellCheckingInspection
-                    if (compareRest(in, "nfinity"))
+                    if (compareRest(in, INFINITY))
                         return Double.POSITIVE_INFINITY;
-                    in.readSkip(-1);
+
                     return Double.NaN;
                 case '-':
+                    negative = true;
+                    in.readSkip(1);
                     if (compareRest(in, INFINITY))
                         return Double.NEGATIVE_INFINITY;
-                    negative = true;
-                    ch = in.rawReadByte();
                     break;
                 default:
                     // do nothing
             }
             int tens = 0;
-            while (true) {
+            while (in.readRemaining() > 0) {
+                ch = in.readUnsignedByte() & 0xFF;
```

Comment:
do we have a performance test for parseDouble method? If not we definitely should have

Reply:
Yes, we do (AppendLongCoolerMain) and it does not show any serious difference:

Before
> PAUSE1,band,<0.1,<1,<10,<100, 1.6, 99.9, 99.9, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.226 / 0.375  0.451 / 0.489  0.538 / 0.702  8.67 / 12.32 - 21.2
> BUSY100,band,<0.1,<1,<10,<100, 83.5, 99.8, 100.0, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.066 / 0.117  0.174 / 0.355  0.598 / 1.89  1.89 / 1.89 - 1.89
> ALL,band,<0.1,<1,<10,<100, 0.0, 99.7, 100.0, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.373 / 0.598  0.690 / 0.822  0.918 / 9.63  9.76 / 9.76 - 9.76

After
> PAUSE1,band,<0.1,<1,<10,<100, 1.7, 99.9, 99.9, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.212 / 0.365  0.431 / 0.481  0.534 / 0.646  8.29 / 9.06 - 21.7
> BUSY100,band,<0.1,<1,<10,<100, 83.1, 100.0, 100.0, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.065 / 0.129  0.234 / 0.391  0.546 / 0.562  0.562 / 0.562 - 0.562
> ALL,band,<0.1,<1,<10,<100, 0.0, 99.4, 100.0, 100.0,%iles,50/90 97/99 99.7/99.9 99.97/99.99 - worst was 0.465 / 0.714  0.826 / 0.942  1.060 / 1.132  1.188 / 1.188 - 1.188

Reply:
love it @alamar